src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -364,196 +359,129 @@ private int RunCore(TextWriter consoleOutput, CancellationToken cancellationToke
-                    using (var win32Res = GetWin32Resources(Arguments, compilation, out errors))
-                    using (pdb)
-                    using (xml)
-                    {
-                        if (PrintErrors(errors, consoleOutput))
-                        {
-                            return Failed;
-                        }
+                finalOutputPath = Path.Combine(Arguments.OutputDirectory, outputName);
+                finalPdbFilePath = Arguments.PdbPath ?? Path.ChangeExtension(finalOutputPath, ".pdb");
 
-                        cancellationToken.ThrowIfCancellationRequested();
+                // NOTE: Unlike the PDB path, the XML doc path is not embedded in the assembly, so we don't need to pass it to emit.
+                var emitOptions = Arguments.EmitOptions.
+                    WithOutputNameOverride(outputName).
+                    WithPdbFilePath(finalPdbFilePath);
 
-                        string outputName = GetOutputFileName(compilation, cancellationToken);
+                using (var emitStreamProvider = new CompilerEmitStreamProvider(this, touchedFilesLogger, finalOutputPath, Arguments.EmitPdb ? finalPdbFilePath : null))
+                {
+                    emitResult = compilation.Emit(emitStreamProvider, xml, win32Res, Arguments.ManifestResources, emitOptions, cancellationToken);
+                }
+            }
 
-                        finalOutputPath = Path.Combine(Arguments.OutputDirectory, outputName);
-                        finalPdbFilePath = Arguments.PdbPath ?? Path.ChangeExtension(finalOutputPath, ".pdb");
+            GenerateSqmData(Arguments.CompilationOptions, emitResult.Diagnostics);
 
-                        // NOTE: Unlike the PDB path, the XML doc path is not embedded in the assembly, so we don't need to pass it to emit.
-                        var emitOptions = Arguments.EmitOptions.
-                            WithOutputNameOverride(outputName).
-                            WithPdbFilePath(finalPdbFilePath);
+            if (PrintErrors(emitResult.Diagnostics, consoleOutput))
+            {
+                return Failed;
+            }
 
-                        emitResult = compilation.Emit(output, pdb, xml, win32Res, Arguments.ManifestResources, emitOptions, cancellationToken);
-                    }
-                }
+            cancellationToken.ThrowIfCancellationRequested();
 
-                GenerateSqmData(Arguments.CompilationOptions, emitResult.Diagnostics);
+            if (analyzerDriver != null)
+            {
+                var analyzerDiagnostics = analyzerDriver.GetDiagnosticsAsync().Result;
+                var allAnalyzerDiagnostics = analyzerDiagnostics.AddRange(analyzerExceptionDiagnostics);
 
-                if (PrintErrors(emitResult.Diagnostics, consoleOutput))
+                if (PrintErrors(allAnalyzerDiagnostics, consoleOutput))
                 {
                     return Failed;
```

Comment:
Yes that is a change.  Need to pass some information down here to let the `Emit` phase know it shouldn't proceed because of the pending anyalzer failures.